sceptre/resolvers/__init__.py

```diff
@@ -56,34 +66,23 @@ def __get__(self, instance, type):
         :return: The attribute stored with the suffix ``name`` in the instance.
         :rtype: dict or list
         """
+        def resolve(attr, key, value):
+            attr[key] = value.resolve()
 
         if hasattr(instance, self.name):
-            return self.resolve_values(getattr(instance, self.name))
+            return _call_func_on_values(
+                resolve, getattr(instance, self.name), Resolver
+            )
 
     def __set__(self, instance, value):
-        setattr(instance, self.name, value)
-
-    def resolve_values(self, attr):
         """
-        Searches through dictionary or list for Resolver objects and replaces
-        them with the resolved value. Supports nested dictionaries and lists.
-        Does not detect Resolver objects used as keys in dictionaries.
+        Attribute setter which adds a stack reference to any resolvers in the
+        data structure `value` and calls the setup method.
 
```

Comment:
Similar to the above suggestion for documenting the types of instance and value.